src/cc/bpf_module_rw_engine.cc

```diff
@@ -400,10 +405,12 @@ int BPFModule::annotate() {
     table_names_[table.name] = id++;
     GlobalValue *gvar = mod_->getNamedValue(table.name);
     if (!gvar) continue;
-    if (PointerType *pt = dyn_cast<PointerType>(gvar->getType())) {
-#if LLVM_MAJOR_VERSION >= 15
-      StructType *st = dyn_cast<StructType>(pt->getPointerElementType());
+#if LLVM_MAJOR_VERSION >= 14
```

Comment:
For consistency, change this to `>= 14`, see https://github.com/llvm/llvm-project/blob/llvmorg-14.0.0/llvm/lib/Transforms/Instrumentation/DataFlowSanitizer.cpp#L249-L258

Reply:
I checked the llvm14 release branch, the below is git blame result:
```
6ff18b08e6bf8 (George Balatsouras      2021-03-02 14:56:16 -0800  249) static StringRef getGlobalTypeString(const GlobalValue &G) {
b7dd329f2f3ff (Alexey Samsonov         2014-07-09 19:40:08 +0000  250)   // Types of GlobalVariables are always pointer types.
5f6eaac6115a4 (Manuel Jacob            2016-01-16 20:30:46 +0000  251)   Type *GType = G.getValueType();
858d38557857b (Eric Christopher        2020-06-20 00:24:57 -0700  252)   // For now we support excluding struct types only.
b7dd329f2f3ff (Alexey Samsonov         2014-07-09 19:40:08 +0000  253)   if (StructType *SGType = dyn_cast<StructType>(GType)) {
b7dd329f2f3ff (Alexey Samsonov         2014-07-09 19:40:08 +0000  254)     if (!SGType->isLiteral())
b7dd329f2f3ff (Alexey Samsonov         2014-07-09 19:40:08 +0000  255)       return SGType->getName();
b7dd329f2f3ff (Alexey Samsonov         2014-07-09 19:40:08 +0000  256)   }
b7dd329f2f3ff (Alexey Samsonov         2014-07-09 19:40:08 +0000  257)   return "<unknown type>";
b7dd329f2f3ff (Alexey Samsonov         2014-07-09 19:40:08 +0000  258) }

```

The changes are all pretty old.

I also checked the patch with a build of llvm14 release branch release/14.x. This patch actually caused the assertion error. With this patch llvm14 is okay.

So I suspect we should guard all changes in this patch with llvm15.

Reply:
OK, just make the changes specific to llvm15.